src/libstore/local-store.cc

```diff
@@ -1431,6 +1433,7 @@ StorePath LocalStore::addToStoreFromDump(Source & source0, std::string_view name
             } else {
                 /* Move the temporary path we restored above. */
                 moveFile(tempPath, realPath);
+                tempDirFd.close();
```

Comment:
Nitpicking a bit, but since it's self closing, I'd rather not close it explicitly, just to keep the accidental complexity out of the way as much as possible